www-client/seamonkey/seamonkey-2.53.11.ebuild

```diff
@@ -0,0 +1,585 @@
+	fi
+
+	if use crypt ; then
+		ewarn
+		ewarn "Enigmail has dropped support for Seamonkey in early 2019. If you are still"
+		ewarn "using it, consider yourself lucky, but also consider to start migrating away"
+		ewarn "from it. The crypt USE flag and its x11-plugins/enigmail dependency will be"
+		ewarn "removed in the near future."
+		ewarn
+	fi
+
+	moz_pkgsetup
+}
+
+pkg_pretend() {
+	# Ensure we have enough disk space to compile
+	if use debug || use lto || use test ; then
+		CHECKREQS_DISK_BUILD="16G"
+	else
+		CHECKREQS_DISK_BUILD="12G"
+	fi
+	check-reqs_pkg_setup
+}
+
+spkg_setup() {
+	# Ensure we have enough disk space to compile
+	if use debug || use lto || use test ; then
+		CHECKREQS_DISK_BUILD="16G"
+	else
+		CHECKREQS_DISK_BUILD="12G"
+	fi
+	check-reqs_pkg_setup
+}
+
+src_unpack() {
+	local l10n_sources="${P}.source-l10n.tar.xz"
+	unpack ${A/ ${l10n_sources}}
+
+	mkdir "${S}/${P}-l10n" || die
+	cd "${S}/${P}-l10n" || die
+	unpack ${l10n_sources}
+}
+
+src_prepare() {
+	SM_PATCHDIR="${WORKDIR}/gentoo-${PN}-patches-${PV}/${PN}"
+	# Apply our patches
+	eapply "${SM_PATCHDIR}/1000_fix-preferences-gentoo.patch"
+	eapply "${SM_PATCHDIR}/1001_gentoo_prefs.patch"
+	eapply "${SM_PATCHDIR}/1002_drop_build_id.patch"
+	eapply "${SM_PATCHDIR}/1003_gentoo_specific_pgo.patch"
+	eapply "${SM_PATCHDIR}/1004_fix_pie_detection.patch"
+	eapply "${SM_PATCHDIR}/1005_fix_fortify_sources.patch"
+	eapply "${SM_PATCHDIR}/1007_re-add_system_sqlite.patch"
+	eapply "${SM_PATCHDIR}/1008_seamonkey-2.53.10.2-remove_gtk2.patch"
+	eapply "${SM_PATCHDIR}/2000_system_harfbuzz.patch"
+	eapply "${SM_PATCHDIR}/2001_system_graphite2.patch"
+	eapply "${SM_PATCHDIR}/2002_bmo-1559213-Support-system-av1.patch"
+	eapply "${SM_PATCHDIR}/2003_nICER_implicit_decls.patch"
+	eapply "${SM_PATCHDIR}/2004_fix_lto_builds.patch"
+	eapply "${SM_PATCHDIR}/2006_musl_sys_auxv.patch"
+	eapply "${SM_PATCHDIR}/2009_upstream-bug_1461221.patch"
+	eapply "${SM_PATCHDIR}/2010_blessings-TERM.patch"
+	eapply "${SM_PATCHDIR}/2011_missing-errno_h-in-SandboxOpenedFiles_cpp.patch"
+	eapply "${SM_PATCHDIR}/2012_make-MOZ_SIGNAL_TRAMPOLINE-Android-only_bug1434526.patch"
+	eapply "${SM_PATCHDIR}/2014_dont_use_gconf_service_mozilla_bug1526243.patch"
+	eapply "${SM_PATCHDIR}/2015_dont_use_gconf_for_proxy_configuration_bug1540145.patch"
+	eapply "${SM_PATCHDIR}/2018_audio_device_backends.patch"
+	eapply "${SM_PATCHDIR}/6001_add_missing_header_for_basename.patch"
+	eapply "${SM_PATCHDIR}/6002_add_alternate_name_for_private_siginfo_struct_member.patch"
+	eapply "${SM_PATCHDIR}/6003_fix_syscall_wrappers_on_musl.patch"
+	eapply "${SM_PATCHDIR}/6004_musl_drop_alloc_hooks.patch"
+	eapply "${SM_PATCHDIR}/6005_musl_memory_report.patch"
+	eapply "${SM_PATCHDIR}/6006_musl_pthread_setname.patch"
+	eapply "${SM_PATCHDIR}/6007_musl_fix_tools.patch"
+	eapply "${SM_PATCHDIR}/6008_musl_fix_toolkit.patch"
+	eapply "${SM_PATCHDIR}/6018_audio_backend_update_generated_files.patch"
```

Comment:
I think `eapply ${WORKDIR}/gentoo-${PN}-patches-${PV}/${PN}/*` would work here. 
Although I'd very much prefer to switch to something similar that firefox, thunderbird and spidermonkey uses at the moment.

Reply:
It's mostly done like this to keep the lines short and ebuild file size reduced.

The way patches were applied in previous ebuilds is the same way firefox, thunderbird and spidermonkey do it, but there is one patch that's USE flag dependent and was patched completely separate from the other patches. I changed to this to keep all the patches together. Some of the other patches in there could also become USE flag dependent, mostly the system-* patches.

Reply:
I don't think you understood what I meant here. Indeed I want to squash these 30 lines into one, and reverse the logic for libvpx patch, same way as is done in firefox+lto. But I can test the syntax and fix-on-merge if that's fine for you.